### DISCUSSION TITLE
[Fooocus-API] The picture sometimes rotates 90 degrees

### DISCUSSION THREAD

**Discussion by Foixa** [lllyasviel/Fooocus#3057] [Q&A] [answered]
Hi, I use Fooocus as an API. It works fine for most images, but sometimes I get a 90 degree rotation. If I try the same image with Fooocus, it gives the correct result.

When I look at the Fooocus logs, I see that it is converting the 3000x4000 image to 4000x3000. The API does not do this.

Can anyone explain exactly what is happening? I have looked through the codes but I have never found a solution.

**Comment by Foixa** [comment]
@mashb1t 

Can you give me a little information about this? Where should I start? I know this problem is not exactly related to Fooocus. Still, it would be nice to have a little information to get me started.

A 3024x4032 photo is processed by fooocus as 4032x3024 and the result is returned as 3024x4032. (This does not happen in every image. only in some images)

I took a look at the source code but couldn't figure out why this was happening. Where should I look?

Thanks

**Comment by Foixa** [comment, marked as answer]
This problem doesn't seem to have anything to do with Fooocus. The problem is related to images converted from Apple Heic format to JPEG.

**Reply by mashb1t** [reply]
@Foixa crazy 🤯 Glad you found the issue, would have looked into this now.

**Reply by Foixa** [reply]
Thank you for your interest.

If someone encounters this kind of issue in the future, I recommend checking the image metadata. The metadata of my input image looked like this:

![Screenshot 2024-05-31 at 18 53 01](https://github.com/lllyasviel/Fooocus/assets/112849271/8148349c-a433-42a5-96c7-67013d6373a6)
